### ISSUE TITLE
nix: jj build failure

### ISSUE THREAD

**Issue by develop7** [martinvonz/jj#299]
## Steps to Reproduce the Problem

1. `nix profile install 'github:martinvonz/jj'`

## Expected Behavior

It is installed successfully

## Actual Behavior

```console
$ nix -v profile install 'github:martinvonz/jj'
building '/nix/store/vkiyx6lphkmn3bg35dhnxlz27c53aig7-jujutsu-unstable-ffc5731.drv'...
error: builder for '/nix/store/vkiyx6lphkmn3bg35dhnxlz27c53aig7-jujutsu-unstable-ffc5731.drv' failed with exit code 101;
       last 10 log lines:
       > Finished cargoSetupPostUnpackHook
       > patching sources
       > Executing cargoSetupPostPatchHook
       > Validating consistency between /build/source//Cargo.lock and /build/cargo-vendor-dir/Cargo.lock
       > Finished cargoSetupPostPatchHook
       > configuring
       > building
       > Executing cargoBuildHook
       > ++ env CC_x86_64-unknown-linux-gnu=/nix/store/pja9g36cy32z3d51942jqk91a6l2d5nv-gcc-wrapper-10.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/pja9g36cy32z3d51942jqk91a6l2d5nv-gcc-wrapper-10.3.0/bin/c++ CC_x86_64-unknown-linux-gnu=/nix/store/pja9g36cy32z3d51942jqk91a6l2d5nv-gcc-wrapper-10.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/pja9g36cy32z3d51942jqk91a6l2d5nv-gcc-wrapper-10.3.0/bin/c++ cargo build -j 24 --target x86_64-unknown-linux-gnu --frozen --release --no-default-features
       > error: package `jujutsu v0.4.0 (/build/source)` cannot be built because it requires rustc 1.60 or newer, while the currently active rustc version is 1.58.1
       For full logs, run 'nix log /nix/store/vkiyx6lphkmn3bg35dhnxlz27c53aig7-jujutsu-unstable-ffc5731.drv'.
```

## Specifications

- Platform: linux x86_64, nix 
- Version: 0.4.0-master

**Comment by develop7** [comment]
My guess would be to run `nix flake update` and commit the updated `flake.lock` file. The nixpkgs got rustc 1.60 already so that should do it.

**Comment by martinvonz** [comment]
Ah, thanks! I looked at that file but didn't see any mention of Rust, so I figured the Rust version was set somewhere else. Turns out it was the whole nixpkgs version that needed to be bumped (which makes sense now that I think about, even with the little I know about Nix). I've sent a fix in #300. I'll also update our CI to use the lock file (it currently overrides it for some reason).
